components/newsletter-sign-up/newsletter-sign-up.jsx

```diff
@@ -0,0 +1,289 @@
+import React from "react";
+import ReactGA from "react-ga";
+import classNames from "classnames";
+import Service from "../../js/service";
+
+class NewsletterSignUp extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      apiSubmitted: false,
+      apiSuccess: false,
+      apiFailed: false,
+      userTriedSubmitting: false
+    };
+  }
+
+  /**
+   * Ensure that the parent component is informed
+   * about this component being mounted (primarily
+   * used in the context of automated testing)
+   */
+  componentDidMount() {
+    if (this.props.whenLoaded) {
+      this.props.whenLoaded();
+    }
+  }
+
+  // state update function
+  apiSubmissionSuccessful() {
+    this.setState({
+      apiSuccess: true
+    });
+  }
+
+  // state update function
+  apiSubmissionFailure(e) {
+    if (e && e instanceof Error) {
+      this.setState({
+        apiFailed: true
+      });
+    }
+  }
+
+  // state check function
+  apiIsDone() {
+    return (
+      this.state.apiSubmitted && (this.state.apiSuccess || this.state.apiFailed)
+    );
+  }
+
+  /**
+   * Submit the user's data to the API server.
+   */
+  submitDataToApi() {
+    this.setState({ apiSubmitted: true });
+
+    return Service.signUp(this.email.value, window.location.toString())
+      .then(() => {
+        this.apiSubmissionSuccessful();
+      })
+      .catch(reason => {
+        console.error(reason);
+        this.apiSubmissionFailure(reason);
+      });
+  }
+
+  /**
+   * Performs very simple validation for emails.
+   * @param {string} input the string that should be validated as an email address
+   * @returns {boolean} true if the input is a legal-enough email address, else false
+   */
+  validatesAsEmail(input) {
```

Comment:
Should we replace this with dummy form field validation?

Reply:
honestly up to you! We can also file a follow-up and get Pomax's opinion coz I'm not 100% sure it's necessarily the right way to do it.